docs/source/data_modeling.rst

```diff
@@ -126,6 +126,96 @@ Query instances
 Sync instances
 ^^^^^^^^^^^^^^^^^
 .. automethod:: cognite.client._api.data_modeling.instances.InstancesAPI.sync
+.. automethod:: cognite.client._api.data_modeling.instances.InstancesAPI.subscribe
+
+Example on syncing instances to local sqlite
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+.. code:: python
+
+    import json
+    import time
+    import sqlite3
+
+    from cognite.client import CogniteClient
+    from cognite.client.data_classes.data_modeling.query import (
+        QueryResult,
+        Query,
+        NodeResultSetExpression,
+        Select,
+    )
+    from cognite.client.data_classes.filters import Equals
+
+
+    DB_NAME = "test.db"
+    SPACE_TO_SYNC = "mytestspace"
+
+    c = CogniteClient()
+
+
+    def sqlite_connection() -> sqlite3.Connection:
+        return sqlite3.connect(DB_NAME, check_same_thread=False)
+
+
+    def sync_space_to_sqlite(space_to_sync: str):
+        with sqlite_connection() as connection:
+            connection.execute(
+                "CREATE TABLE IF NOT EXISTS instance (space TEXT, external_id TEXT, data JSON)"
+            )
+            connection.execute(
+                "CREATE INDEX IF NOT EXISTS space_extid_idx ON instance (space, external_id)"
+            )
+            connection.execute("CREATE TABLE IF NOT EXISTS cursor (cursor TEXT)")
+            existing_cursor = connection.execute(
+                "SELECT cursor FROM cursor"
+            ).fetchone()
+
+        if existing_cursor:
+            print("Found existing cursor, using that")
+
+        query = Query(
+            with_={
+                "nodes": NodeResultSetExpression(
+                    filter=Equals(property=["node", "space"], value=space_to_sync)
+                )
+            },
+            select={"nodes": Select()},
+            cursors={"nodes": existing_cursor[0] if existing_cursor else None},
+        )
+
+        def _sync_batch_to_sqlite(result: QueryResult):
+            with sqlite_connection() as connection:
+                inserts = [
+                    (node.space, node.external_id, json.dumps(node.dump()))
+                    for node in result["nodes"]
+                    if node.deleted_time is None
+                ]
+                deletes = [
+                    (node.space, node.external_id)
+                    for node in result["nodes"]
+                    if node.deleted_time is not None
+                ]
+                # Updates must be done in the same transaction as persisting the cursor.
+                # A transaction is implicitly started by sqlite here.
+                connection.executemany(
+                    "INSERT INTO instance VALUES (?, ?, ?)", inserts
+                )
+                connection.executemany(
```

Comment:
Delete must happen before insert. The same `space, external_id` may appear as several tombstones and then a new instance, which must result in the instance being saved.

Reply:
Ah right. TY 🙏